runtime/src/identity.rs

```diff
@@ -589,6 +637,106 @@ decl_module! {
                 Err("Account cannot remove this authorization")
             }
         }
+
+
+        /// It adds signing keys to target identity `id`.
+        /// Keys are directly added to identity because each of them has an authorization.
+        ///
+        /// Arguments:
+        ///     - `origin` Master key of `id` identity.
+        ///     - `id` Identity where new signing keys will be added.
+        ///     - `additional_keys` New signing items (and their authorization data) to add to target
+        ///     identity.
+        ///
+        /// Failure
+        ///     - It can only called by master key owner.
+        ///     - Keys should be able to linked to any identity.
+        pub fn add_signing_items_with_authorization( origin,
+                id: IdentityId,
+                expires_at: T::Moment,
+                additional_keys: Vec<SigningItemWithAuth>) -> Result {
+            let sender = ensure_signed(origin)?;
+            let sender_key = Key::try_from(sender.encode())?;
+            let _grants_checked = Self::grant_check_only_master_key(&sender_key, id)?;
+
+            // 0. Check expiration
+            let now = <timestamp::Module<T>>::get();
+            ensure!( now < expires_at, "Offchain authorization has expired");
+            let authorization = TargetIdAuthorization {
```

Comment:
Sorry, @fmiguelgarcia I didn't get the use of this authorization struct and how it will help in the process?

Reply:
It is the data that we verify to be signed.

Reply:
ahh! The high tone of the naming convention makes me confuse.